dvc/utils/__init__.py

```diff
@@ -152,9 +152,9 @@ def makedirs(path, exist_ok=False, mode=None):
         _makedirs(path, exist_ok=exist_ok)
         return
 
-    umask = os.umask(0)
+    umask = os.umask(0o777 - mode)
```

Comment:
Let's put a comment explaining what is going on :)

Reply:
👍